src/blueapi/worker/task_worker.py

```diff
@@ -177,12 +223,15 @@ def mark_task_as_started(event: WorkerEvent, _: str | None) -> None:
             if (
                 event.task_status is not None
                 and event.task_status.task_id == trackable_task.task_id
+                and trackable_task.task_id in self._context_register
             ):
                 task_started.set()
 
         LOGGER.info(f"Submitting: {trackable_task}")
         try:
             sub = self.worker_events.subscribe(mark_task_as_started)
+            self._context_register[trackable_task.task_id] = get_trace_context()
```

Comment:
Must: Context register does not ever appear to be cleared, so if blueapi is left running for weeks or months (quite likely) it will get quite big and cumbersome. This class is already complicated so I'm not immediately sure if introducing extra bookkeeping is the correct solution. Maybe the context information could be embedded in the documents themselves?

Reply:
or clear the context for a task when it finishes

Reply:
Clearing on finish sounds like a reasonabe idea

Reply:
Added code to pop the entry from the dictionary to _cycle after the task has finished.